packages/opentelemetry-api/src/api/propagation.ts

```diff
@@ -104,6 +107,25 @@ export class PropagationAPI {
     delete _global[GLOBAL_PROPAGATION_API_KEY];
   }
 
+  /**
+   * Adds a propagator factory under a unique name
+   */
+  public registerNamedPropagator(
```

Comment:
What is the point of constructing and registering propagators by name? Shouldn't the sdk simply look at the environment variable and construct only what is required? This seems wasteful and add surface to the API package.

Reply:
I can't require a package without it being automatically added to a front-end bundle (e.g. in webpack).

Reply:
Another way to solve it would be for propagator packages to rely on another package , so they could register themselves automatically, but I thought we rejected that idea in bitter.

Reply:
I don't understand. In the `TracerProvider` you just have code like this:

```javascript
import {B3Propagator} from "wherever-b3-comes-from";

// get propagator constructs the appropriate propagator based on the environment config
// Falls back to the default propagator
function getPropagator() {
	// switch might not work depending on how the env var works.
	// might have to use if/else blocks
	switch(getEnv(propagatorEnvVar)) {
	  case "b3":
	    return new B3Propagator();
	  case "jaeger":
		// etc
	  default:
	    return new CompositePropagator(traceContext, baggage);
	}
}
```

Reply:
We also talked about optionally allowing to register these in the browser version, hence the `registerNamedPropagator` function. Spec also describes supporting other non-core propagators, and we could not do that, but a registering function seemed fitting at the time. Anyway, let me split this code into a separate PR and continue the discussion there.